pandas/core/dtypes/dtypes.py

```diff
@@ -666,11 +660,15 @@ def construct_from_string(cls, string):
         raise TypeError("could not construct PeriodDtype")
 
     def __unicode__(self):
-        return "period[{freq}]".format(freq=self.freq.freqstr)
+        return self.name
 
     @property
     def name(self):
-        return str(self)
+        return u"period[{freq}]".format(freq=self.freq.freqstr)
+
+    @property
+    def na_value(self):
+        return iNaT
```

Comment:
What's the logic for why iNaT instead of NaT?

Reply:
It should be NaT. I got the user-facing and physical-storage semantics backwards.